### ISSUE TITLE
Option to simplify Debts

### ISSUE THREAD

**Issue by KMKoushik** [oss-apps/split-pro#58]

**Comment by krokosik** [comment]
I would be happy to work on this and started looking at how can this be implemented.

My main inspiration is of course Splitwise and I would implement the maximum flow algorithm for optimization, as described [here](https://medium.com/@mithunmk93/algorithm-behind-splitwises-debt-simplification-feature-8ac485e97688). However, currently the app does not really seem to have a very compatible UI for that.

In particular, the UI currently does not really isolate groups on the main screen as Splitwise does and it is only that screen that shows balances. The debt simplification algorithm could of course only be applied to groups, as we cannot affect every single user on Split Pro with our decision to simplify debts (I know a lot of people who dislike/distrust this feature).

What I would propose is to go along with the more group oriented design, as this is the core functionality of such apps and the primary use case as far as my experience goes. However, I would like to have a green light from the maintainer @KMKoushik, as I do not know if you want to go in this direction or have a different vision for this.

To summarize the changes I would like to implement:
- display our givers and receivers in the group info, instead of just the total amount owed
- add a field to group details called simplifyDebts
- add a toggle in group settings to switch simplify debts (any member can do it, not just the owner)
- introduce a per-peson balance query, which either calculates individual in group transfers or performs simplified ones

**Comment by tamtam22** [comment]
Is there any timeline to accept the pull request?

**Comment by krokosik** [comment]
The repo isnot maintained currently, you can use my fork to get this feature and I would appreciate the feedback :)

**Comment by Soulplayer** [comment]
Is there a docker image i could pull? @krokosik

**Comment by krokosik** [comment]
Sorry, I was hoping to upstream my changes, so I only built it locally

**Comment by Soulplayer** [comment]
@krokosik is this now in the main version? I would love to be able to simplify the debts.

**Comment by krokosik** [comment]
@Soulplayer Yes

**Comment by Soulplayer** [comment]
I do not see this option in group page? Is this setting hidden somewhere?

![image](https://picsur.verhaeghe-steyaert.be/i/cbe9cdd3-3fcf-4fd7-965e-ed6cb00c8e52.jpg)

**Comment by krokosik** [comment]
That's not the version from the main branch. Did you build the image yourself?

**Comment by Soulplayer** [comment]
No I use docker. So I’m on that latest version. I pull it from the suggested repo

**Comment by krokosik** [comment]
I still haven't released the 1.4 version. It will be included there.

**Comment by krokosik** [comment]
Hey, I've released a pre-release image in case you want to try the feature: https://github.com/oss-apps/split-pro/releases/tag/v1.4.0-beta.0

**Comment by Soulplayer** [comment]
Thank you. This worked!
